utils.py

```diff
@@ -0,0 +1,97 @@
+import argparse
+import torch
+import numpy as np
+
+ATOL = 1e-5
+RTOL = 1e-5
+
+def is_model_on_hammerblade(model):
+    return next(model.parameters()).is_hammerblade
+
+def argparse_inference():
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('--nbatch', default=1, type=int,
+                        help="Number of batches to be tested")
+    parser.add_argument('--hammerblade', default=False, action='store_true',
+                        help="Run on HammerBlade")
+    parser.add_argument('--batch_size', default=1, type=int,
+                        help="Training batch size")
+    parser.add_argument("--dry", default=False, action='store_true',
+                        help="Dry run")
+    parser.add_argument('--filename', default="trained_model", type=str,
+                        help="Filename of the saved model")
+    return parser.parse_args()
+
+def argparse_training():
```

Comment:
I really like the idea of having a common argparsing method. just wondering how should we handle workload specific options. e.g., `Rescyc` has a `validate` option.

Maybe we can have a common method called `add_common_options` (just like gem5's way if you are familiar)

Reply:
I get your concern... instead of the common method doing `return parser.parse_args()`, it can do `return parser`.

Does this sounds good? We could add workload specific options to parser this way...

Reply:
I ended up doing it in a slightly different way. The common argument parsing method takes in a function which will injects workload specific options. so we can keep code like this in common space
https://github.com/cornell-brg/pytorch-apps/blob/2b7768a350acf635084e8414cbf33a856266d3d7/utils.py#L56-L59

Reply:
The expected behavior is: if either one is specified, the one not being specified will not run.

--training -> training only
--inference -> inference only
NONE -> both

Reply:
Sounds good!